addons/website_mass_mailing/models/res_company.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    def _get_social_media_links(self):
+        self.ensure_one()
+        website_id = self.env['website'].get_current_website()
+        return {
+            'social_facebook': website_id.social_facebook or self.social_facebook,
+            'social_linkedin': website_id.social_linkedin or self.social_linkedin,
+            'social_twitter': website_id.social_twitter or self.social_twitter,
+            'social_instagram': website_id.social_instagram or self.social_instagram
+        }
```

Comment:
You should call super, then check website_id and update values with website_id.social_* if any.
So we do not mess up with hypothetical future overrides